src/eckit/cmd/CMakeLists.txt

```diff
@@ -1,85 +1,94 @@
 list( APPEND eckit_cmd_srcs
-AliasCmd.cc
-AliasCmd.h
-Arg.cc
-Arg.h
-CmdApplication.cc
-CmdApplication.h
-CmdArg.cc
-CmdArg.h
-CmdParser.cc
-CmdParser.h
-CmdResource.cc
-CmdResource.h
-ConfigCmd.cc
-ConfigCmd.h
-DirCmd.cc
-DirCmd.h
-EchoCmd.cc
-EchoCmd.h
-EnvironmentCmd.cc
-EnvironmentCmd.h
-ExportCmd.cc
-ExportCmd.h
-JSONCmd.cc
-JSONCmd.h
-HistoryCmd.cc
-HistoryCmd.h
-KillCmd.cc
-KillCmd.h
-LibEcKitCmd.cc
-LibEcKitCmd.h
-LockCmd.cc
-LockCmd.h
-ManCmd.cc
-ManCmd.h
-MemoryCmd.cc
-MemoryCmd.h
-PsCmd.cc
-PsCmd.h
-QuitCmd.cc
-QuitCmd.h
-RemoteCmd.cc
-RemoteCmd.h
-RemoteCommandable.cc
-RemoteCommandable.h
-RemoteCommander.cc
-RemoteCommander.h
-RemoteCommandUser.cc
-RemoteCommandUser.h
-SleepCmd.cc
-SleepCmd.h
-StartCmd.cc
-StartCmd.h
-StatusCmd.cc
-StatusCmd.h
-StopCmd.cc
-StopCmd.h
-TailCmd.cc
-TailCmd.h
-TermBuf.cc
-TermBuf.h
-term.c
-UpTimeCmd.cc
-UpTimeCmd.h
-UserInput.h
-UserInput.cc
+    AliasCmd.cc
+    AliasCmd.h
+    Arg.cc
+    Arg.h
+    CmdApplication.cc
+    CmdApplication.h
+    CmdArg.cc
+    CmdArg.h
+    CmdParser.cc
+    CmdParser.h
+    CmdResource.cc
+    CmdResource.h
+    ConfigCmd.cc
+    ConfigCmd.h
+    DirCmd.cc
+    DirCmd.h
+    EchoCmd.cc
+    EchoCmd.h
+    EnvironmentCmd.cc
+    EnvironmentCmd.h
+    ExportCmd.cc
+    ExportCmd.h
+    HistoryCmd.cc
+    HistoryCmd.h
+    JSONCmd.cc
+    JSONCmd.h
+    KillCmd.cc
+    KillCmd.h
+    LibEcKitCmd.cc
+    LibEcKitCmd.h
+    LockCmd.cc
+    LockCmd.h
+    ManCmd.cc
+    ManCmd.h
+    MemoryCmd.cc
+    MemoryCmd.h
+    PsCmd.cc
+    PsCmd.h
+    QuitCmd.cc
+    QuitCmd.h
+    RemoteCmd.cc
+    RemoteCmd.h
+    RemoteCommandUser.cc
+    RemoteCommandUser.h
+    RemoteCommandable.cc
+    RemoteCommandable.h
+    RemoteCommander.cc
+    RemoteCommander.h
+    SleepCmd.cc
+    SleepCmd.h
+    StartCmd.cc
+    StartCmd.h
+    StatusCmd.cc
+    StatusCmd.h
+    StopCmd.cc
+    StopCmd.h
+    TailCmd.cc
+    TailCmd.h
+    TermBuf.cc
+    TermBuf.h
+    UpTimeCmd.cc
+    UpTimeCmd.h
+    UserInput.cc
+    UserInput.h
+    term.c
 )
 
-ecbuild_generate_yy( YYPREFIX eckit_cmd_
-                     YACC cmdsy
-                     LEX  cmdsl
-                     DEPENDANT CmdParser.cc )
+ecbuild_generate_yy(
+    YYPREFIX eckit_cmd_
+    YACC cmdsy
+    LEX  cmdsl
+    DEPENDANT CmdParser.cc
+)
 
-if( NOT ENABLE_WARNINGS )
-  # Warnings present in autogenerated files based on cmdsy and cmdsl
-  list( APPEND suppress_warnings
-    $<$<CXX_COMPILER_ID:GNU>:-Wno-unused-function>
-    $<$<CXX_COMPILER_ID:GNU>:-Wno-sign-compare>
-    $<$<CXX_COMPILER_ID:Clang>:-Wno-deprecated-register>
-  )
-  set_source_files_properties( CmdParser.cc PROPERTIES COMPILE_OPTIONS "${suppress_warnings}" )
+# Supress warnings emited from YACC generated code
+if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
```

Comment:
This looks like a change in functionality. It has lost the if (NOT ENABLE_WARNINGS) test. Is that intentional?